crates/rpc/src/handlers/block.rs

```diff
@@ -272,3 +279,46 @@ pub async fn get_block_from_id(
 
     Ok(HttpResponse::Ok().json(BeaconVersionedResponse::new(beacon_block)))
 }
+
+/// Called by `/beacon/heads` to get fork choice leaves.
+#[get("/beacon/heads")]
+pub async fn get_beacon_heads(db: Data<ReamDB>) -> Result<impl Responder, ApiError> {
+    let base = db
+        .justified_checkpoint_provider()
+        .get()
+        .map_err(|_| ApiError::InternalError)?;
+    let root = base.root;
+
+    let mut blocks = HashMap::with_hasher(RandomState::default());
```

Comment:
```suggestion
    let mut blocks = HashMap::new();
```

Reply:
I've tried clear out the types but it cause error on applying params in `filter_block_tree`.

<img width="923" alt="image" src="https://github.com/user-attachments/assets/b2df9b98-b39e-46a4-bde6-36961dde8fe4" />

Reply:
Instead of using alloy_primitives can you switch the hashmap we use to Hashbrown https://crates.io/crates/hashbrown

Reply:
I've tried earlier. Still the the error since `filter_block_tree` used `HashMap` from `alloy_primitives`
<img width="680" alt="image" src="https://github.com/user-attachments/assets/6447dbca-cf34-4ddf-bcd3-d385e76a0f34" />

Reply:
I would remove all instances in the codebase of `HashMap` from `alloy_primiatives` and replace it with `HashMap` from `HashBrown`

Reply:
I left a comment here if you could resolve it https://github.com/ReamLabs/ream/pull/337#pullrequestreview-2861863001